R/notin.R

```diff
@@ -0,0 +1,6 @@
+# Intended to be used to create %notin% operator
+notin = function(example, elements) {
+  return(!match(example, elements, nomatch = 0))
+}
+
+"%notin%" = function(example, elements) notin(example, elements)
```

Comment:
Same branching here -- we wrote an api for `%chin%` that gives the `%chin%` output directly instead of wrapping `chmatch() > 0L`

Reply:
@MichaelChirico Could you explain this a little more? I was trying to mimic the implementation of `%like%` where there is a defined R function `like` that's wrapped make the operator. So if we have branching for characters in the `notin` function, then shouldn't that handle branching for the opreator as well?

Reply:
`notin` does handle branching, but it dispatches to `chmatch`. If we branch in `%notin%`, we can dispatch to `%chin%` directly (which doesn't use `chmatch` in R).

This is different for character input because if we dispatch to `%in%`, it will just re-route to `match` anyway

Reply:
I'm sorry if I'm missing something, but it seems like if it's better to dispatch to `%chin%` than `chmatch`, then we should use `%chin%` in both `notin` and `%notin%` and then their definitions would be the same.

This is would be the operator implementation and the function would match:


```
"%notin%" = function(example, elements) {
  if (is.character(example)) {
    return(!(example %chin% elements))
  } else {
    return(!match(example, elements, nomatch = 0))
  }
}
```

Reply:
I think that makes sense. Actually I am not sure if we need `notin()` -- for `like()`, it's a more extensible version of `%like%` since it allows to control several more options. whereas here, the signature is the same, it's just a difference of using it as an operator. 

FWIW, I think we should also look into trying to add an option in the C API like `invert` so that we don't need to `!` at the end . Then the function could look like 

```
"%notin%" = function(example, elements) {
  if (is.character(example)) {
    return(.Call(cchin, x, table, /* invert= */TRUE))
  } else {
    return(!match(example, elements, nomatch = 0))
  }
}
```

Reply:
That sounds like a good idea to me. I will try to change the `chin` function to include the invert parameter. It looks like this means altering `chmatchMain` which is also used for `chmatch` and `chmatchdup`, so those would also be easy to negate.

Reply:
I think I've implemented this change pretty efficiently. Please let me know if this looks good or if there are any suggestions.